### PR TITLE
Documentation/revise view

### PR DESCRIPTION
**Pull request by ajpowelsnl**
Minor edits to View to stage for conversion to new API standard entry.

### CODE REVIEW

File: docs/source/API/core/view/view.rst [hunk 1]
```diff
@@ -1,65 +1,64 @@
...
     :language: cppkokkos
 
 Header File: ``<Kokkos_Core.hpp>``
 
-Usage
------
+Description
+--------------
+
+The programming model's main data structure is the Kokkos View.  The View has multi-dimensional (up to eight) array-like semantics and operations.  It, and related data structures (*e.g.*, `View-like Types <view_like.html>`__)  can be used on the CPU (host) and/or GPU (device), and has architecture-specific optimiztions, shared-memory parallelism, compile-time layouts and memory space.
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:12]: This is mostly not quite right. View does not have array-like semantics or operations actually. It also doesn't have shared-memory parallelism per-se, and I wouldn't necessarily call it the programming model's main data structure - which has some unfortunate connotations.

File: docs/source/API/core/view/view.rst [hunk 2]
```diff
@@ -1,65 +1,64 @@
...
-Its semantics are similar to that of ``std::shared_ptr``.
 
 Interface
----------
+------------
 
 .. code-block:: cpp
 
-    template <class DataType [, class LayoutType] [, class MemorySpace] [, class MemoryTraits]>
+    template <class DataType [, class Space] [, class Layout] [, class MemoryTraits]>
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:20]: Where is this coming from? It was correct before now its wrong?

File: docs/source/API/core/view/view.rst [hunk 3]
```diff
@@ -1,65 +1,64 @@
...
 
-.. |LayoutLeft| replace:: ``LayoutLeft()``
+.. |LayoutLeft| replace:: :cppkokkos:func:``LayoutLeft``
 
 .. _LayoutStride: layoutStride.html
 
-.. |LayoutStride| replace:: ``LayoutStride()``
+.. |LayoutStride| replace:: :cppkokkos:func:``LayoutStride``
+
+The ``DataType`` template parameter is required, but the others are optional.  Ordering is enforced for only the first (``DataType``) and last (``MemoryTraits``) positions.  That means, for example, that ``Layout`` can be omitted, but if both ``MemorySpace`` and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTraits``.  Ordering is therefore not strongly enforced for the ``Space`` or ``ArrayLayout`` parameters.
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:38]: This sentence is wrong. If you provide both Space and Layout, Layout has to come first.

File: docs/source/API/core/view/view.rst [hunk 4]
```diff
@@ -68,35 +67,35 @@ and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTra
...
 
 .. _Atomic: ../atomics.html
 
 .. |Atomic| replace:: :cppkokkos:func:`Atomic`
 
 
 * ``MemoryTraits``:
-
-  Sets access properties via enum parameters for the templated ``Kokkos::MemoryTraits<>`` class.
+  MemoryTraits specify how a data structure is accessed.
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:79]: It is not a "data structure" which is accessed, but elements of the View, which more often than not are not structures in any C++ sense.

File: docs/source/API/core/view/view.rst [hunk 5]
```diff
@@ -68,35 +67,35 @@ and ``MemoryTraits`` are specified, ``MemorySpace`` must come before ``MemoryTra
...
+  Sets access properties via enum parameters for the templated ``Kokkos::MemoryTraits<>`` class.  
   Possible template parameters are bit-combinations of the following flags:
 
   - ``Unmanaged``: The View will not be reference counted. The allocation has to be provided to the constructor.
 
-  - |Atomic|_: All accesses to the view will use atomic operations.
+  - |Atomic|_: All accesses to the View will use atomic operations.
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:85]: if we capitalize we should make it code font. Note that "view" is a general word referring to what `View` is a: a data structure with "view" semantics.

File: docs/source/API/core/view/view.rst [hunk 6]
```diff
@@ -108,10 +107,12 @@ instead of relying on implicit conversion to an integral type.
 
 The actual type of ``rank[_dynamic]`` as it was defined until Kokkos 4.1 was left up to the implementation
 (that is, up to the compiler not to Kokkos) but in practice it was often ``int`` which means
-this change may yield warnings about comparing signed and unsigned integral types.
+this change may yield warnings about comparing signed and unsigned integral types.0
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:110]: random zero?

File: docs/source/API/core/view/view.rst [hunk 7]
```diff
@@ -295,31 +322,27 @@ Constructors
 
 .. cppkokkos:function:: View( pointer_type ptr, const IntType& ... indices)
 
-   Unmanaged data wrapping constructor.
+   Unmanaged data-wrapping constructor.
 
-   - ``ptr``: pointer to a user provided memory allocation. Must provide storage of size ``View::required_allocation_size(n0,...,nR)``
+   - ``ptr``: pointer to a user-provided memory allocation. Must provide storage of size ``View::required_allocation_size(n0,...,nR)`` or 
+              ``View::required_allocation_size(layout)``.
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:328]: Whats going on here? There is no layout parameter above? Did you try to merge the two constructors??

File: docs/source/API/core/view/view.rst [hunk 8]
```diff
@@ -295,31 +322,27 @@ Constructors
...
+   - ``layout``: an instance of a layout class. The number of valid extents must
+     either match the dynamic rank or the total rank. In the latter case, the extents
+     corresponding to compile-time dimensions must match the View type's compile-time extents.
+
    - Requires: ``sizeof(IntType...)==rank_dynamic()`` or ``sizeof(IntType...)==rank()``. In the latter case,
      the extents corresponding to compile-time dimensions must match the View type's compile-time extents.
 
    - Requires: ``array_layout::is_regular == true``.
 
 .. cppkokkos:function:: View( pointer_type ptr, const array_layout& layout)
```
Comment (crtrott) [inline docs/source/API/core/view/view.rst:341]: This constructor has now no description?

**Review by crtrott — CHANGES_REQUESTED** [review]
This can't be merged there are a whole bunch of mistakes in this which need fixing.

**Comment by ajpowelsnl** [comment]
No need for a wholesale revision of `View` documentation at this time; instead, make incremental changes (together with code updates) as needed.

**Pull request closed without merging**